geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -187,6 +188,224 @@ object GeoMesaSpark extends LazyLogging {
     groupedByDay.map { case (date, iter) => (date, iter.size) }
   }
 
+  def shallowJoin(sc: SparkContext, coveringSet: RDD[SimpleFeature], data: RDD[SimpleFeature], key: String): RDD[SimpleFeature] = {
+    // Broadcast sfts to executors
+    val sfts = sc.broadcast(GeoMesaSparkKryoRegistrator.typeCache.values.map{ sft =>
+      (sft.getTypeName, SimpleFeatureTypes.encodeType(sft))
+    }.toArray)
+
+    data.foreachPartition{ iter =>
+      sfts.value.foreach{ case (name, spec) =>
+        val sft = SimpleFeatureTypes.createType(name, spec)
+        GeoMesaSparkKryoRegistrator.putType(sft)
+      }
+    }
+
+    // Broadcast covering set to executors
+    val broadcastedCover = sc.broadcast(coveringSet.collect)
+
+    // Key data by cover name
+    val keyedData = data.mapPartitions { iter =>
+      import org.locationtech.geomesa.utils.geotools.Conversions._
+
+      iter.flatMap { sf =>
+        // Iterate over covers until a match is found
+        val it = broadcastedCover.value.iterator
+        var container = ""
+        while (it.hasNext && container == "") {
+          val cover = it.next()
+          // If the cover's polygon contains the feature, set the container
+          if (cover.geometry.contains(sf.geometry)) {
```

Comment:
I think `intersects` may be a bit more generally applicable and would be _almost_ equivalent in the case of a polygon covering set and point data. The difference would be points on the boundary of a polygon would count for the polygon.

The intersect is really interesting and useful for `data` with LineString or Polygon geometries. In this case one expects an individual data to possibly `intersect  more than one covering set. And plausibly `contain`ed by none.